schema/artists/popular.js

```diff
@@ -24,9 +24,8 @@ const PopularArtists = {
       description: "Number of results to return",
     },
   },
-  resolve: ({ userID, popularArtistsLoader, authenticatedPopularArtistsLoader }, options) => {
-    const loader = userID ? authenticatedPopularArtistsLoader : popularArtistsLoader
-    return loader(options)
+  resolve: ({ popularArtistsLoader }, options) => {
```

Comment:
Is this `resolve` function special in that it receives the loaders as the first param rather than the 4th as with the other resolvers?

Reply:
How interesting: 
```
{
  popular_artists {
    artists {
      id
    }
  }
}
```
Worked just fine so I just assumed? Why do you think this works? I can move to `rootValue: { popularArtistsLoader }` but that leads to the question -- is this a better way to pull loaders out?

Reply:
Nice! 
```js
resolve: async ({ saleArtworksFilterLoader }, options) => {
    const params = parseRelayOptions({ ...DEFAULTS, ...options })
    const response = await saleArtworksFilterLoader(params)
```
worked just fine.

Reply:
Interesting! We’ve actually always been using the wrong parameter to store these things like `accessToken` and loaders in, those are really supposed to go into the 3rd parameter which is the ‘context’. What you are discovering here is that the `rootValue` object is actually meant to set values that are always available in the 1st param (called the `root`) which otherwise only holds data from the ‘parent’ type.

Nonetheless, I think we should keep using the 4th parameter, because we can then more easily fix all of the incorrect usage at some point.

Reply:
👍 That will be a _very_ nice refactor, ditching all those additional args

Reply:
Put different, the `root` should contain data that may be queryable, such as data coming from an API.